src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -2645,21 +2848,32 @@ void CallStubGenerator::ProcessArgument(ArgIterator *pArgIt, ArgLocDesc& argLocD
     {
         argType = RoutineType::Stack;
     }
-    
+
     TerminateCurrentRoutineIfNotOfNewType(argType, pRoutines);
 
     if (argLocDesc.m_cGenReg != 0)
     {
 #if LOG_COMPUTE_CALL_STUB
         printf("m_cGenReg=%d\n", (int)argLocDesc.m_cGenReg);
 #endif // LOG_COMPUTE_CALL_STUB
+#ifdef TARGET_ARM
+        if (splitRequired)
+        {
+            pRoutines[m_routineIndex++] = GetRegRoutine_4B(argLocDesc.m_idxGenReg, argLocDesc.m_idxGenReg + argLocDesc.m_cGenReg - 1);
+        }
+        else
+#endif // TARGET_ARM
         if (m_r1 == NoRange) // No active range yet
         {
             // Start a new range
             m_r1 = argLocDesc.m_idxGenReg;
             m_r2 = m_r1 + argLocDesc.m_cGenReg - 1;
         }
+#ifdef ENREGISTERED_PARAMTYPE_MAXSIZE
         else if (argLocDesc.m_idxGenReg == m_r2 + 1 && (!pArgIt || !pArgIt->IsArgPassedByRef()))
+#else
+        else if (argLocDesc.m_idxGenReg == m_r2 + 1)
+#endif // ENREGISTERED_PARAMTYPE_MAXSIZE
```

Comment:
A nit - can you please keep only the extra part of the condition under the ifdef here? Like this:
```suggestion
        else if (argLocDesc.m_idxGenReg == m_r2 + 1 
#ifdef ENREGISTERED_PARAMTYPE_MAXSIZE
        && (!pArgIt || !pArgIt->IsArgPassedByRef())
#endif
        )
```